test/e2e/sagemaker/tests/test_endpoint.py

```diff
@@ -0,0 +1,264 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the SageMaker Endpoint API.
+"""
+
+import boto3
+import pytest
+import logging
+import time
+from typing import Dict
+
+from sagemaker import (
+    SERVICE_NAME,
+    service_marker,
+    CRD_GROUP,
+    CRD_VERSION,
+    CONFIG_RESOURCE_PLURAL,
+    MODEL_RESOURCE_PLURAL,
+    ENDPOINT_RESOURCE_PLURAL,
+)
+from sagemaker.replacement_values import REPLACEMENT_VALUES
+from common.resources import load_resource_file, random_suffix_name
+from common import k8s
+
+
+@pytest.fixture(scope="module")
+def sagemaker_client():
+    return boto3.client("sagemaker")
+
+
+@pytest.fixture(scope="module")
+def _create_endpoint():
```

Comment:
nit: fixture name should represent the mocks they create. In this case maybe `single_variant_xgboost_endpoint`?